CONTRIBUTING.md

```diff
@@ -0,0 +1,234 @@
+
+- *accessibility* generator: Use Git to review the differences between the `cypress-accessibility` branch and the `cypress-baseline` branch. Again creating the relevant unit tests.
+
+***Step 3: Implementation***
+
+It is worth reviewing existing plugins to understand how we utilise the `@nx/devkit` and `@ensono-stacks/core` to create and manipulate files.
+
+Some important initial checks:
+
+1. `verifyPluginCanBeInstalled()`: Checks to see if the workspace is compatible
+
+2. `hasGeneratorExecutedForProject()`: If the generator can only be ran once, this ensures that the `nx.json` and stacks `executedGenerators` list is correctly updated.
+
+Once the code has been written for your generators, you must create corresponding e2e tests to ensure that your plugin can be built, published and executed within a stacks/nx workspace.
+
+## Testing
+
+### Unit Testing
+
+Unit tests should looks to cover all areas of a generator in regards to file creation and manipulation. By following a TDD approach this should ensure maximum coverage for all of the files which are affected by a generator.
+
+- You can use the Jest runner plugin in Visual Studio Code or Nx to run the tests with the following commands:
+
+  - Test a specific package
+
+  ```bash
+  nx test <package-name>
+  ```
+
+  - Test all packages that have been updated
+
+  ```bash
+  nx affected -t test
+  ```
+
+  - Run the test target for multiple packages
+
+  ```bash
+  nx run-many -t test -p proj1 proj2
+  ```
+
+  - Run the test target for all packages
+
+  ```bash
+  nx run-many --target test --all --parallel 8
+  ```
+
+### End to end testing
+
+End to end tests look to cover usage of the plugin within an Nx workspace, in order verify this we have built a custom [e2e executor](./packages/common/e2e/src/executors/e2e/executor.ts) to run our tests using [verdaccio](https://verdaccio.org/) as a private proxy registry.
+
+The following are the steps in which e2e tests follow:
+
+1. The e2e executor builds, packages and deploys the selected plugin and it's dependent packages, including the create and workspace packages, to the verdaccio instance.
+2. The [newProject(stacksPackageToInstall, nxPackagesToInstall)](./packages/common/e2e/src/utils/project.ts) method will then create a new Stacks workspace using the `@ensono-stacks/create-stacks-workspace` plugin which has been published to verdaccio. It will then install the plugin you are testing in the workspace.
+3. With the workspace provisioned, e2e tests should look to verify that the various generators available to the plugin can be executed and utilised within the workspace.
+
+To run your e2e tests you must use the e2e executor:
+
+```bash
+nx e2e <plugin name>-e2e
+```
+
+### Testing packages locally
+
+- Option 1: Use npm linking
+  - TO BE COMPLETED - @Metecan

```

Comment:
Are we ok to have this in the main branch without the notes about npm linking?

Reply:
Updated with linking guide